tests/mqtt-frames-truncated/test.rules

```diff
@@ -0,0 +1,17 @@
+alert mqtt any any -> any any (msg:"mqtt Frame 1"; frame:pdu; content:"|10 1c|"; startswith; sid:1;)
+alert mqtt any any -> any any (msg:"mqtt Frame 2"; frame:pdu; content:"|14|"; endswith; sid:2;)
+
+alert mqtt any any -> any any (msg:"mqtt Frame 3"; frame:header; content:"|10|"; sid:3;)
+alert mqtt any any -> any any (msg:"mqtt Frame 4"; frame:header; content:"|10 1c|"; sid:4;)
+
+alert mqtt any any -> any any (msg:"mqtt Frame 5"; frame:pdu; content:"|17 0C E2|"; sid:5;)
+alert mqtt any any -> any any (msg:"mqtt Frame 6"; frame:pdu; content:"|00 00 54 46|"; sid:6;)
+
+# pre-boundary test for truncated data
+alert mqtt any any -> any any (msg:"mqtt Frame 7"; frame:data; content:"|0a|"; sid:7;)
+
+# At boundary test for truncated data
+alert mqtt any any -> any any (msg:"mqtt Frame 8"; frame:data; content:"|51|"; sid:8;)
```

Comment:
Following the conversation from the last PR. ref https://github.com/OISF/suricata-verify/pull/1172#discussion_r1168263677

I meant that one byte is generally not a unique indicator of the data.
1. Does `|51|` only occurs once in the entire pcap? 
      - Yes. Then sure, this is good.
      - No. Then, how do you know that the alert happened at the correct `|51|` byte? A unique sequence is usually a much better indicator. So, you could have say three or two bytes before this `|51|` to make sure it is the exact sequence that you intended to match with.